### PR TITLE
Remove deprecated packager environment variables

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.org/wp/45102
https://community.openproject.org/wp/43551

### CODE REVIEW

File: packaging/conf/configuration.yml [hunk 1]
```diff
@@ -35,9 +35,6 @@
 #################################################################
 
 default:
-  rails_cache_store: <%= ENV.fetch('RAILS_CACHE_STORE') { :memcache }.to_sym %>
-  session_store: <%= ENV.fetch('SESSION_STORE') { :active_record_store }.to_sym %>
```
Comment (oliverguenther) [inline packaging/conf/configuration.yml:39]: SESSION_STORE is never set by packager and the given value is already the default. So if it is used by someone without prefix, they can just replace it themselves

**Comment by cbliard** [comment]
I started a packager build to test it: https://packager.io/gh/opf/openproject/builds/6562

File: packaging/addons/openproject/bin/postinstall [hunk 2]
```diff
@@ -63,16 +63,19 @@ rake_commands="db:migrate db:seed"
...
 else
-  ${CLI} config:set RAILS_CACHE_STORE=memcache
+  ${CLI} config:set OPENPROJECT_RAILS__CACHE__STORE=memcache
 fi
 
+${CLI} config:unset RAILS_CACHE_STORE
+
 # create attachments folder
-attachments_path=$(${CLI} config:get ATTACHMENTS_STORAGE_PATH || echo "/var/db/${APP_NAME}/files")
+attachments_path=$(${CLI} config:get OPENPROJECT_ATTACHMENTS__STORAGE__PATH || ${CLI} config:get ATTACHMENTS_STORAGE_PATH || echo "/var/db/${APP_NAME}/files")
```
Comment (cbliard) [inline packaging/addons/openproject/bin/postinstall:74]: Documentation needs to be updated too: `docs/installation-and-operations/configuration/README.md:286` has a paragraph about running `openproject config:get ATTACHMENTS_STORAGE_PATH` to customize the path.

Also, I do not know if `OPENPROJECT_ATTACHMENTS_STORAGE_PATH` should be supported too.

Comment (cbliard) [reply packaging/addons/openproject/bin/postinstall:74]: There are other references to ATTACHMENTS_STORAGE_PATH:
* `docs/installation-and-operations/misc/migration/README.md`
* `packaging/scripts/backup`
* and 2 others involving docker in `docker/prod/Dockerfile` and `docker/prod/entrypoint.sh`

File: packaging/conf/configuration.yml [hunk 3]
```diff
@@ -35,9 +35,6 @@
 #################################################################
 
 default:
-  rails_cache_store: <%= ENV.fetch('RAILS_CACHE_STORE') { :memcache }.to_sym %>
```
Comment (cbliard) [inline packaging/conf/configuration.yml:38]: There are some references to `RAILS_CACHE_STORE` in documentation (related to docker) and in some docker-related files. Not sure if it should be updated too or not.

**Review by cbliard — CHANGES_REQUESTED** [review]
It works, the deprecation warnings do not appear anymore. 

There is some documentation that needs to be updated as well, and some usage of `RAILS_CACHE_STORE` and `ATTACHMENTS_STORAGE_PATH` in docker-related files for which I do not what should be done. Please see comments.

File: packaging/conf/configuration.yml (hunk 3 above)
Comment (oliverguenther) [reply packaging/conf/configuration.yml:38]: I've opened https://community.openproject.org/work_packages/45416 for the Dockerfile, as there are more envs judging by a quick look at the dockerfile.

File: packaging/addons/openproject/bin/postinstall (hunk 2 above)
Comment (cbliard) [reply packaging/addons/openproject/bin/postinstall:74]: Fixed in 3553a80990.

File: packaging/conf/configuration.yml (hunk 3 above)
Comment (cbliard) [reply packaging/conf/configuration.yml:38]: Fixed in 3553a80990.

File: docker/prod/entrypoint.sh [hunk 4]
```diff
@@ -44,9 +44,13 @@ if [ "$(id -u)" = '0' ]; then
 	  rm -f $APACHE_PIDFILE || true
 	fi
 
-	if [ ! -z "$ATTACHMENTS_STORAGE_PATH" ]; then
-		mkdir -p "$ATTACHMENTS_STORAGE_PATH"
-		chown -R "$APP_USER:$APP_USER" "$ATTACHMENTS_STORAGE_PATH"
+	# Use ATTACHMENTS_STORAGE_PATH value when OPENPROJECT_ATTACHMENTS__STORAGE__PATH is not set
+	OPENPROJECT_ATTACHMENTS__STORAGE__PATH=${OPENPROJECT_ATTACHMENTS__STORAGE__PATH:-$ATTACHMENTS_STORAGE_PATH}
```
Comment (crohr) [inline docker/prod/entrypoint.sh]: I think it should be `export`ed

Comment (cbliard) [reply docker/prod/entrypoint.sh]: You're right, thanks. I have amended the previous commit.

**Pull request merged**